plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java

```diff
@@ -244,14 +247,24 @@ public void put(final InputStream inputStream) throws IOException, NugetPackageE
       recordMetadata.put(CREATED, creationTime);
       recordMetadata.put(LAST_UPDATED, creationTime);
       recordMetadata.put(PUBLISHED, creationTime);
-
+      OrientVertex component = null;
       try (InputStream in = tempStream.get()) {
-        createOrUpdatePackage(storageTx, recordMetadata, in);
+        component = createOrUpdatePackage(storageTx, recordMetadata, in);
       }
 
-      maintainAggregateInfo(storageTx, recordMetadata.get(ID));
+      String id = recordMetadata.get(ID);
+      String version = recordMetadata.get(VERSION);
+      maintainAggregateInfo(storageTx, id);
 
+      boolean isNew = component.getIdentity().isNew();  // must check before commit
       storageTx.commit();
+
+      if (isNew) {
+        getEventBus().post(new ComponentCreatedEvent(component, getRepository()));
+      }
+      else {
+        getEventBus().post(new ComponentUpdatedEvent(component, getRepository()));
+      }
     }
```

Comment:
Yeah, since the message encapsulates the vertex, and the message is acted upon synchronously, a current OrientDB connection must be owned by the current thread. Ideally, firing these events occurs just after the transaction is committed but before the OrientDB connection (encapsulated by the StorageTx object), is returned to the pool. And that looks like what's happening.